storage/src/main/java/org/apache/kafka/storage/internals/log/ProducerStateManagerConfig.java

```diff
@@ -17,13 +17,12 @@
 package org.apache.kafka.storage.internals.log;
 
 import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.coordinator.transaction.TransactionLogConfigs;
 
 import java.util.Set;
 
 public class ProducerStateManagerConfig {
-    public static final String PRODUCER_ID_EXPIRATION_MS = "producer.id.expiration.ms";
-    public static final String TRANSACTION_VERIFICATION_ENABLED = "transaction.partition.verification.enable";
-    public static final Set<String> RECONFIGURABLE_CONFIGS = Utils.mkSet(PRODUCER_ID_EXPIRATION_MS, TRANSACTION_VERIFICATION_ENABLED);
+    public static final Set<String> RECONFIGURABLE_CONFIGS = Utils.mkSet(TransactionLogConfigs.PRODUCER_ID_EXPIRATION_MS_CONFIG, TransactionLogConfigs.TRANSACTION_PARTITION_VERIFICATION_ENABLE_CONFIG);
```

Comment:
Do we need to put those in `ProducerStateManagerConfig`? `DynamicBrokerConfig` is the class that has power to decide the reconfigurable configs (https://github.com/apache/kafka/blob/f895ab5145077c5efa10a4a898628d901b01e2c2/core/src/main/scala/kafka/server/DynamicBrokerConfig.scala#L1136).  and so It seems to me those reconfigurable configs should be moved to `DynamicBrokerConfig`.

Reply:
I think this is to follow the same pattern we had `DynamicBrokerConfig .AllDynamicConfigs` that it refer to list of `RECONFIGURABLE_CONFIGS` from each module But I also can see that we kinda broke this pattern in 
`DynamicBrokerConfig.DynamicSecurityConfigs`, `DynamicBrokerConfig.DynamicPasswordConfigs` and `DynamicBrokerConfig.ReloadableFileConfigs`. I'll move to DynamicBrokerConfig to follow the same pattern for DynamicSecurityConfigs